### ISSUE TITLE
An update that does not request any fields fails

### ISSUE THREAD

**Issue by harendra-kumar** [haskell-graphql/graphql-api#174]
A pure update that does not request any fields in the results must be allowed. The following query fails to parse:

```
{SetResource(id:"1",resource:{resAge:20,resGender:"m"}){}}
```

**Comment by jml** [comment]
Huh, interesting. I think you’re right, but can you point to relevant bit
of the spec? Will make it easier to fix.

**Comment by harendra-kumar** [comment]
I did not look at the spec. I just thought it makes sense and assumed it. I will update if I happen to take a look at the spec.

**Comment by harendra-kumar** [comment]
yeah, I was confused whether to use `{}` or just omit it for an empty request, that must get clarified the spec.

**Comment by jml** [comment]
AFAICT, the GraphQL spec does not prohibit empty selection sets, so this is a valid bug. However, it also does not say how to handle them, so fixing involves some decision making. 

Things that you or someone else can do to help fix this bug:

- review the [spec](http://facebook.github.io/graphql/draft/#) to see if I've missed something
- identify what other (esp. Facebook-authored) GraphQL servers do with this
  - include steps to reproduce, versions, etc.
- write a test that would pass if this bug were fixed
